librz/include/rz_util/rz_path.h

```diff
@@ -2,13 +2,22 @@
 #define RZ_UTIL_PATH_H_
 
 #include <rz_types.h>
+#include <rz_th.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-RZ_API void rz_path_set_prefix(RZ_NONNULL const char *path);
-RZ_API RZ_OWN char *rz_path_prefix(RZ_NULLABLE const char *path);
+typedef struct rz_path_t {
+	char *prefix;
+	bool prefix_searched;
+	RzThreadLock *prefix_mutex;
+} RzPath;
+
+#ifdef RZ_API
+
+RZ_API void rz_path_set_prefix(RzPath *sys_path, RZ_NONNULL const char *path);
```

Comment:
add RZ_NONNULL like for path and check the pointer in the assert (rz_return_if_fail) with the function.